File: lib/scripts/ruby/bundler.rb
module Ruby
  extend self
  def bundler
    unless installed?("bundler")
      command("gem", "install", "bundler", "--no-ri", "--no-rdoc")
    end

    install_path = env_fetch("BUNDLER_INSTALL_PATH", "vendor")

    unless test_command?("bundler", "check")
      command("bundler", "install", "--path", install_path)
    end
  end
end

File: internal/scripts/ruby/bundler.go
package CIScriptsRuby

import (
	c "github.com/bcaldwell/ci-scripts/internal/CIScriptsHelpers"
)

type Bundler struct{}

func (b *Bundler) Run() error {
	if !c.CheckBinary("bundler") {
		err := c.Command("gem", "install", "bundler", "--no-ri", "--no-rdoc")
		if err != nil {
			return err
		}
	}

	installPath, _ := c.ConfigFetch("ruby.bundler.install_path", "vendor")

	if !c.TestCommand("bundler", "check") {
		err = c.Command("bundler", "install", "--path", installPath)
		if err != nil {
			return err
		}
	}

	return nil
}
